orderbook/usecase/orderbook_usecase_test.go

```diff
@@ -401,42 +457,144 @@ func (s *OrderbookUsecaseTestSuite) TestGetActiveOrders() {
+				withGetMetadataByChainDenom(tokensusecase, "quoteToken")
+			},
+			expectedError: &types.FailedToGetMetadataError{},
+		},
+		{
+			name: "failed to get base token metadata",
+			setupMocks: func(usecase *orderbookusecase.OrderbookUseCaseImpl, orderbookrepository *mocks.OrderbookRepositoryMock, client *mocks.OrderbookGRPCClientMock, tokensusecase *mocks.TokensUsecaseMock) {
+				client.GetActiveOrdersCb = s.getActiveOrdersFunc(orderbookdomain.Orders{s.newOrder()}, 1, nil)
+				withGetMetadataByChainDenom(tokensusecase, "quoteToken")
+			},
+			expectedError: &types.FailedToGetMetadataError{},
+		},
+		{
+			name: "error on creating formatted limit order ( no error - best effort )",
+			setupMocks: func(usecase *orderbookusecase.OrderbookUseCaseImpl, orderbookrepository *mocks.OrderbookRepositoryMock, client *mocks.OrderbookGRPCClientMock, tokensusecase *mocks.TokensUsecaseMock) {
+				client.GetActiveOrdersCb = s.getActiveOrdersFunc(orderbookdomain.Orders{s.newOrder()}, 1, nil)
+				withGetMetadataByChainDenom(tokensusecase, "")
+				orderbookrepository.GetTickByIDFunc = s.getTickByIDFunc(orderbookdomain.OrderbookTick{}, false)
+			},
+			expectedOrders:       []orderbookdomain.LimitOrder{},
+			expectedError:        nil,
+			expectedIsBestEffort: true,
+		},
+		{
+			name: "successful processing of active orders",
+			setupMocks: func(usecase *orderbookusecase.OrderbookUseCaseImpl, orderbookrepository *mocks.OrderbookRepositoryMock, client *mocks.OrderbookGRPCClientMock, tokensusecase *mocks.TokensUsecaseMock) {
+				client.GetActiveOrdersCb = s.getActiveOrdersFunc(orderbookdomain.Orders{s.newOrder()}, 1, nil)
+				withGetMetadataByChainDenom(tokensusecase, "")
+				orderbookrepository.GetTickByIDFunc = s.getTickByIDFunc(s.newTick("500", 100, "bid"), true)
+				tokensusecase.GetSpotPriceScalingFactorByDenomFunc = s.getSpotPriceScalingFactorByDenomFunc(1, nil)
+			},
+			expectedError: nil,
+			expectedOrders: []orderbookdomain.LimitOrder{
+				s.newLimitOrder(),
+			},
+			expectedIsBestEffort: false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			// Create instances of the mocks
+			client := mocks.OrderbookGRPCClientMock{}
+			tokensusecase := mocks.TokensUsecaseMock{}
+			orderbookrepository := mocks.OrderbookRepositoryMock{}
+
+			// Setup the mocks according to the test case
+			usecase := orderbookusecase.New(&orderbookrepository, &client, nil, &tokensusecase, &log.NoOpLogger{})
+			if tc.setupMocks != nil {
+				tc.setupMocks(usecase, &orderbookrepository, &client, &tokensusecase)
+			}
+
+			order := s.newLimitOrder()
+
+			// Call the method under test
+			orders, isBestEffort, err := usecase.ProcessOrderBookActiveOrders(context.Background(), domain.CanonicalOrderBooksResult{
+				ContractAddress: order.OrderbookAddress,
+				PoolID:          1,
+				Quote:           order.QuoteAsset.Symbol,
+				Base:            order.BaseAsset.Symbol,
+			}, "ownerAddress")
```

Comment:
Similar to comments in other PRs, I think we should have one test case where we vary the `ownderAddress`